### ISSUE TITLE
Unexpected results for `str_starts()` when use pattern containing `"|"`

### ISSUE THREAD

**Issue by psychelzh** [tidyverse/stringr#449]
See the following example:

``` r
stringr::str_starts("A1", "T")
#> [1] FALSE
stringr::str_starts("A1", "\\d")
#> [1] FALSE
stringr::str_starts("A1", "T|\\d")
#> [1] TRUE
```

<sup>Created on 2022-07-31 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by psychelzh** [comment]
Also found this [SO](https://stackoverflow.com/a/71426424/5996475). So perhaps  it is a documentation issue?

**Comment by psychelzh** [comment]
Ah, I am sorry. This might have been solved by #340. So when will there be a new release?

**Comment by hadley** [comment]
Not sure. Hopefully in the next couple of months.
